test/main/modules.js

```diff
@@ -49,6 +49,33 @@ QUnit.test( "does not run before subsequent tests", function( assert ) {
 	assert.equal( this.beforeCount, 1, "beforeCount did not increase from last test" );
 } );
 
+QUnit.module( "before (skip)", {
+	before: function( assert ) {
+		assert.ok( true, "before hook ran" );
+
+		if ( typeof this.beforeCount === "undefined" ) {
+			this.beforeCount = 0;
+		}
+
+		this.beforeCount++;
+	}
+} );
+
+QUnit.skip( "skipped, but should still run before hook", function( assert ) {
+	assert.expect( 1 );
+	assert.ok( true );
+} );
+
+QUnit.test( "has run by the time second (unskipped) test runs", function( assert ) {
+	assert.expect( 1 );
+	assert.equal( this.beforeCount, 1, "beforeCount should be one" );
```

Comment:
if we remove the `this.assert = new Assert( this );` for skipped tests line will this fail?